### PR TITLE
chore(deps): fix Dependabot alerts in sandbox

### PR DESCRIPTION
**Pull request by cploujoux**
Dependabot sweep for sandbox: **11 alerts closed (1 critical, 4 high, 5 medium, 1 low) / 1 high skipped (no patch exists) / 0 blocked / 0 deferred**. All 12 alerts are one package in one file.

Two evidence bars. Tier A (critical/high) requires a per-package test that runs in CI; tier B (moderate/low) requires the gate green plus a recorded reachability check. Here the package carries a critical, so the whole row is held to tier A. Live alert list matches the snapshot (12 alerts), re-checked after committing.

| package | manifest | old → new | severity | tier | advisories | reachable from prod? | test |
|---|---|---|---|---|---|---|---|
| nltk | hub/jupyter-server/requirements.txt | 3.10.0 → 3.10.3 | critical + 4 high + 5 medium + 1 low | A | GHSA-m4rf-3fr8-xwx3 (critical), GHSA-qx2g-xrx7-vfh8, GHSA-6hwm-xvph-95vm, GHSA-5gh2-94qg-qppq (high), 5 medium, 1 low. GHSA-8mgp-746c-j5xp (high) has no fix. | yes: `textblob==0.20.1` in the same file requires nltk>=3.9, and notebook users can import it. No repo code imports it. | `hub/jupyter-server/tests/test_nltk_security.py`, 12 new tests, runs in the existing gate job `jupyter-server-python-deps` |

**Target version**: 3.10.3 is the max fix across the ten patchable alerts and the newest nltk on PyPI. OSV cross-check: 24 records match 3.10.0, all but GHSA-8mgp-746c-j5xp cleared by 3.10.3, including 13 advisories Dependabot does not list.

**Validation**
- Image built with the bump; inside it: 181 resolved packages, exactly one moved (nltk). `pip check` clean.
- The 12 new tests are hermetic (no network, no JVM, temp dirs only). 10 of them fail on 3.10.0 and pass on 3.10.3, verified by running the suite against both versions. Two examples: the critical (dangerous JVM flags such as `-agentlib:jdwp` and `-javaagent:` now raise before spawning) and GHSA-5gh2-94qg-qppq (a pickle gadget `sklearn.os.system` that actually executes on 3.10.0 is rejected on 3.10.3).
- Six medium/low advisories rest on the version floor plus the OSV audit: no distinguishable behavioural signature was found for them. The test module docstring says so.

**Verification gate**
Unchanged. `dependency-verification.yml` already lists both touched paths in its filter and already matches this branch prefix, so the job runs (not skipped) and picks up the new tests.

**Behaviour changes for notebook users (nltk 3.10.3 upstream)**
- `-XX:*` and `-D*` java options are now rejected. Escape hatch: `java(..., trusted_raw_options=[...])`.
- A new classpath sandbox requires jars to sit inside an `nltk_data` root. The image ships none, so pointing at a jar elsewhere raises `UntrustedJarError`. Escape hatch: `NLTK_ALLOW_UNSAFE_JARS=1`.
- The image has no JVM, so neither is reachable until a user installs a JDK themselves. Worth a line in the image release notes.
- `AllowlistUnpickler` rejects more names (dunder, numpy load/save, pandas.read_pickle). `TweetTokenizer` no longer matches more than 126 labels as one URL.

**Not fixed**
- GHSA-8mgp-746c-j5xp (high): path handling in `TransitionParser`, `AveragedPerceptron` and tagger save/load bypasses `nltk.pathsec`. No patched version exists (GitHub says `<= 3.10.3`, OSV's PYSEC record says fixed in 3.10.3, treated as open). Preconditions are pathsec enforcement plus untrusted model paths, neither of which this image sets up.

**After merge**
This change is under hub/, not sandbox-api/, so `release.yaml` does not fire. See the workflow triggers listed in the PR conversation for what a merge to develop or main builds.

---

<a href="https://app.devin.ai/review/blaxel-ai/sandbox/pull/317" target="_blank">
  <picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://static.devin.ai/assets/gh-devin-review-dark.svg?v=3">
    Devin Review
  </picture>
</a>

### CODE REVIEW

**Comment by cploujoux** [comment]
Workflow triggers for this change (hub/jupyter-server only):
- `dependency-verification.yml` runs on this PR and picks up the 12 new tests (its paths filter lists both touched files).
- `build.yaml` fires on push to main and develop and rebuilds the hub images, which is the real end-to-end proof of the bump on Linux.
- `release.yaml` and `test.yaml` are filtered on sandbox-api/** and do not fire.

File: hub/jupyter-server/tests/test_nltk_security.py [hunk 1]
```diff
@@ -0,0 +1,382 @@
...
+
+    from nltk.parse.dependencygraph import dot2img
+    from nltk.translate.api import AlignedSent
+
+    with pytest.raises(Exception, match="Cannot find the dot binary"):
+        dot2img("digraph G { a -> b }")
+    assert not marker.exists(), "dot2img executed the planted ./dot"
+
+    with pytest.raises(Exception, match="Cannot find the dot binary"):
+        AlignedSent(["a"], ["b"])._repr_svg_()
```
[the remark this answers is not in the document]
Comment (cploujoux) [reply hub/jupyter-server/tests/test_nltk_security.py]: Agreed. The test now asserts the security property only (the planted ./dot is never executed, checked through the marker file) and expects the 'Cannot find the dot binary' error only when no real dot exists in /usr/bin or /bin. With Graphviz installed, nltk running the real binary or failing on the input both pass. 11 passed, 1 skipped locally on 3.10.3 without Graphviz.
